chainer/datasets/cifar.py

```diff
@@ -123,13 +123,13 @@ def creator(path):
             # training set
             for i in range(5):
                 file_name = '{}/data_batch_{}'.format(dir_name, i + 1)
-                d = pickle.load(archive.extractfile(file_name))
+                d = pickle.load(archive.extractfile(file_name), encoding='latin-1')
```

Comment:
pickle.load only takes `encoding` keyword in python3. How about defining a function that chooses the right function to call based on python's version:
`def pickle_load(f):
    if sys.version_info > (3, ):
        # python3
        return pickle.load(f, encoding='latin-1')
    else:
        # python2
        return pickle.load(f)`